### DISCUSSION TITLE
Incorrect js new Date() object.

### DISCUSSION THREAD

**Discussion by dimitarnikolovv** [vercel/community#5124] [Help] [answered]
### Question

I have a serverside code that should create a `new Date()` object based on frontend user input and convert it to ISO format.
The problem is that when tested locally it works fine, but in production the datetime object is incorrect.

This is the code: 

```js
    console.log(data.departDate, data.departTime);

    data.departDateTime = new Date(data.departDate);
    data.departDateTime.setHours(data.departTime.split(':')[0]);
    data.departDateTime.setMinutes(data.departTime.split(':')[1]);
    data.departDateTime.toISOString();

    console.log('DDT:', data.departDateTime);
```

This is the expected output:
![Screenshot 2023-12-07 at 8 12 36](https://github.com/vercel/vercel/assets/49731322/1ab1ada2-350d-49f1-a105-4e3de11fc2a2)

This is what is outputted in vercel:
![Screenshot 2023-12-07 at 8 13 29](https://github.com/vercel/vercel/assets/49731322/21c32116-e3ea-4576-bb46-ae9be7f8caeb)

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out, @dimitarnikolovv. I moved your question from the `vercel` repo to our community support page.

This looks like a timezone issue. When you run it locally, the dev server and the frontend are both using your local timezone. But, when you deploy it, the server uses UTC time. Please make sure the dates you send to the function use UTC time and that should solve it 🙏
